src/components/CPicker/CPicker.js

```diff
@@ -118,8 +125,9 @@ const getTimePickerListeners = (context) => {
       const splitTime = value.split(':');
       const hours = splitTime[0];
       const minutes = splitTime[1];
-      const formattedValue = moment.utc(self.value).hours(hours).minutes(minutes).toISOString();
-
+      const formattedValue = setMinutes(setHours(zonedTimeToUtc(self.value
+        ? parseISO(self.value) : new Date(), Intl.DateTimeFormat().resolvedOptions().timeZone),
+      parseInt(hours, 10)), parseInt(minutes, 10)).toISOString();
```

Comment:
1. Can we use [set](https://date-fns.org/v2.7.0/docs/set) instead of separate `setMinutes` and `setHours` methods?
2. Don't use ternary operator as a function argument, we lose code readability, separate it in a helper variable if this logic is necessary (is it even necessary to parse to ISO format incoming value?)

Reply:
1. I will change it to set
2.I will simplify it
3.I just converted functions from moment.js to date-fns without changing the format like the issue stated. I suppose there is no need for zonedTimeToUtc, is this the case with parsedTimeValue function?

Reply:
Sorry, I see now that this local variable `formattedValue` is actually assigned to `self.value` and isn't the same as `self.formattedValue` 😓 You can disregard third item in my comment.